src/plugins/discover/public/application/main/hooks/use_text_based_query_language.ts

```diff
@@ -67,11 +67,9 @@ export function useTextBasedQueryLanguage({
           const { index, viewMode } = stateContainer.appState.getState();
           let nextColumns: string[] = [];
           const isTextBasedQueryLang =
-            recordRawType === 'plain' &&
-            isOfAggregateQueryType(query) &&
-            ('sql' in query || 'esql' in query);
+            recordRawType === 'plain' && isOfAggregateQueryType(query) && 'esql' in query;
```

Comment:
```suggestion
            recordRawType === 'plain' && isOfAggregateQueryType(query);
```
Nit: not sure this is needed anymore either.

Reply:
[7dcf660](https://github.com/elastic/kibana/pull/182079/commits/7dcf660596ac845511abebc8adda264c6ead53d4)